app/upgrades/v16/upgrades.go

```diff
@@ -134,25 +135,38 @@ func CreateUpgradeHandler(
 
 		// Get community pool and DAI/OSMO pool address.
 		communityPoolAddress := keepers.AccountKeeper.GetModuleAddress(distrtypes.ModuleName)
-		daiOsmoPool, err := keepers.PoolManagerKeeper.GetPool(ctx, DaiOsmoPoolId)
+
+		// Determine the amount of OSMO that can be bought with 1 DAI.
+		oneDai := sdk.NewCoin(DAIIBCDenom, sdk.NewInt(1000000000000000000))
+		daiOsmoGammPool, err := keepers.PoolManagerKeeper.GetPool(ctx, DaiOsmoPoolId)
 		if err != nil {
 			return nil, err
 		}
-
-		// Swap in a maximum of 10 OSMO for one DAI via the community pool.
-		oneDai := sdk.NewCoin(DAIIBCDenom, sdk.NewInt(1000000000000000000))
-		tokenInAmt, err := keepers.GAMMKeeper.SwapExactAmountOut(ctx, communityPoolAddress, daiOsmoPool, DesiredDenom0, sdk.NewInt(10000000), oneDai, sdk.ZeroDec())
+		respectiveOsmo, err := keepers.GAMMKeeper.CalcOutAmtGivenIn(ctx, daiOsmoGammPool, oneDai, DesiredDenom0, sdk.ZeroDec())
 		if err != nil {
 			return nil, err
 		}
 
 		// Create a full range position via the community pool with the funds that were swapped.
-		fullRangeOsmoDaiCoins := sdk.NewCoins(sdk.NewCoin(DesiredDenom0, tokenInAmt), oneDai)
+		fullRangeOsmoDaiCoins := sdk.NewCoins(respectiveOsmo, oneDai)
 		_, _, _, _, err = keepers.ConcentratedLiquidityKeeper.CreateFullRangePosition(ctx, clPoolId, communityPoolAddress, fullRangeOsmoDaiCoins)
 		if err != nil {
 			return nil, err
 		}
 
+		// Because we are doing a direct send from the community pool, we need to manually change the fee pool to reflect the change.
+
+		// Remove coins we used from the community pool to make the CL position
+		feePool := keepers.DistrKeeper.GetFeePool(ctx)
+		newPool, negative := feePool.CommunityPool.SafeSub(sdk.NewDecCoinsFromCoins(fullRangeOsmoDaiCoins...))
+		if negative {
+			return nil, fmt.Errorf("community pool cannot be negative: %s", newPool)
+		}
+
+		// Update and set the new fee pool
+		feePool.CommunityPool = newPool
+		keepers.DistrKeeper.SetFeePool(ctx, feePool)
```

Comment:
Could you please add tests to check that the community pool balance as well as this `feePool` are updated in the `upgrades_test.go`?

Reply:
Test added! https://github.com/osmosis-labs/osmosis/pull/5659/commits/4cece9a3c0b23c6d77995c72d1b6f06faadc2823